added calendar and date selection for tasks

vinograd1kaa opened the pull request.

vinograd1kaa committed:
added calendar and date selection for tasks

vinograd1kaa committed:
added calendar icon to tasks

vinograd1kaa committed:
added a calendar to each todoItem

vinograd1kaa committed:
changed item date is confirmed after closing

vinograd1kaa committed:
1.fixed double click
2.added sorting
3.fixed calendar component

vinohradov commented on src/components/Calendar/Calendar.jsx:
```diff
@@ -0,0 +1,16 @@
+import { CalendarContainer } from './styles/index';
+
+const Calendar = ({ calendarDate, handleClickCalendarDay }) => {
+  const [value, onChange] = useState(calendarDate || new Date());
```
выглядит так как будто тебе не нужен этот компонент и ты можешь юзать напрямую компонент из библиотеки

vinohradov replied:
стили календаря есть в npm пакете календаря, импортируй от туда
если это не styled-components то стили подключаются в /src/globalStyles

vinohradov commented on src/containers/Todo/TodoItem.jsx:
```diff
@@ -111,6 +112,29 @@ const TodoItem = ({
+      .map((el) => el);
+
+    if (
+      date.day === dateLetters[2] &&
```
это что за проверка?

vinohradov commented on src/containers/Todo/TodoItem.jsx:
```diff
@@ -13,9 +13,20 @@ import {
+  itemChecked,
+  itemNotToMove,
+  itemIdToMove,
+  calendarItem,
```
переписать все названия нужно и называть одинаково:
тебе не нужно указывать везде слово `item`. 
Так как это и так понятно потому что у тебя компонент называется TodoItem
calendarItem - плохое название, лучше isCalendarOpen
itemParentId - parentId
и так далее

vinohradov replied:
держи isCalendarOpen внутри каждого таска

vinohradov commented on src/containers/Todo/TodoItem.jsx:
```diff
@@ -172,13 +220,23 @@ const TodoItem = ({ items, itemParentId, itemChecked, itemNotToMove, itemIdToMov
               itemChecked={itemChecked || item.isChecked}
               itemParentId={item.id}
               itemIdToMove={itemIdToMove}
               itemNotToMove={itemNotToMove || item.id === itemIdToMove}
```
тебе ж не нужна эта переменная, ты ее всегда можешь вычислить на лету

vinohradov commented on src/reducers/todo/index.js:
```diff
@@ -47,6 +70,11 @@ export default function todoReducer(state = initialState, { type, payload }) {
             isExpended: false,
             isChecked: false,
             parentId: payload.id,
+            date: state.items[payload.id].date || {
```
для работы с датой используй `new Date().getTime()`
тебе не нужно вычислять день месяц и год каждый раз

vinohradov commented on src/reducers/todo/index.js:
```diff
@@ -100,28 +144,56 @@ export default function todoReducer(state = initialState, { type, payload }) {
+        if (
+          parentTask &&
+          (parentTask.date.day !== currentTask.date.day ||
+            parentTask.date.month !== currentTask.date.month ||
```
непонятно зачем тут это большое условие

vinohradov commented on src/reducers/todo/index.js:
```diff
@@ -100,28 +144,56 @@ export default function todoReducer(state = initialState, { type, payload }) {
         items: { ...state.items },
       };
 
+    case CALENDAR_TASK:
```
плохое название экшна которое не говорит не о чем

vinograd1kaa committed:
fixed all the shortcomings that sensei wrote about

vinograd1kaa committed:
fixed time bugs

vinograd1kaa committed:
deleted unnecessary files)

vinograd1kaa committed:
fix all bugs and add header

vinograd1kaa committed:
now I don’t pass idNotTomove (replaced it with a function for 20 lines to make it beautiful)

vinohradov commented on src/containers/Todo/TodoItem.jsx:
```diff
@@ -13,16 +12,20 @@ import {
+import Calendar from '../../components/Calendar/index';
 
-const TodoItem = ({ items, itemParentId, itemChecked, itemNotToMove, itemIdToMove }) => {
+const TodoItem = ({ items, id, title, isCalendarOpen, isExpanded, isChecked, parentId }) => {
```
items бери из useSelector

vinohradov replied:
календарь у тебя не работает

vinograd1kaa committed:
1.added funk getSubTasks (utils)
2.fixed bug with date
3.fixed bug with checked subTasks
4.fixed bug with localstorage
5.added blur on calendar

vinograd1kaa committed:
fix bugs

vinograd1kaa committed:
added moment library and date logic

vinograd1kaa committed:
fixed bug with localStorage and useEffect

vinograd1kaa committed:
fixed big with rerender useEffect
